### PR TITLE
use single icon file + provide generic icon component

### PR DESCRIPTION
**Pull request by martinklepsch**
trello: https://trello.com/c/tZqW1kc8/361-svg-image-issues

Goal: replace most icons with icons loaded from a single cachable file.

___

Review with https://github.com/open-company/open-company-api/pull/37

___

To test: run locally, 

- check that all topics have correct icons. (You may need to re-import sample data.)
- check that topic edit UI has checkboxes + that they work
- check that topic overlay has icons in top right (pencil, close)

### CODE REVIEW

**Comment by bago2k4** [comment]
reviewing...

File: src/open_company_web/components/icon.cljs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  Optional second argument can be used to pass a map of options:
+  - :accent-color (oc-blue)
+  - :size (30px)
+  - :stroke (2px)"
+  ([id] (icon id {}))
+  ([id {:keys [accent-color size stroke] :as opts}]
+   (assert id "Icon ID is required")
+   (let [outline-color "#0f0f0f"
+         accent-color  (or accent-color "#007A9D") ; oc-blue from scss/modules/_colors.scss
```
Comment (bago2k4) [inline src/open_company_web/components/icon.cljs]: @martinklepsch we should import this color from `open-company-web.oc-colors`, you can get it with
```clojure
(open-company-web.oc-colors/get-color-by-kw :oc-blue-light)
```

File: src/open_company_web/components/icon.cljs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  Screenshot of Download window: http://i.imgur.com/ltAj6X1.png
+
+  Optional second argument can be used to pass a map of options:
+  - :accent-color (oc-blue)
+  - :size (30px)
+  - :stroke (2px)"
+  ([id] (icon id {}))
+  ([id {:keys [accent-color size stroke] :as opts}]
+   (assert id "Icon ID is required")
+   (let [outline-color "#0f0f0f"
```
Comment (bago2k4) [inline src/open_company_web/components/icon.cljs]: @martinklepsch also this color should be added to oc-colors

**Comment by martinklepsch** [comment]
@bago2k4 updated

**Comment by bago2k4** [comment]
@martinklepsch i think something broke here, if you try to add a topic (click on the manage topic button and then click on a random topic, save). The server respond with:
```
Not processable: {:customer-service {:image missing-required-key}, :mission {:image missing-required-key}, :update {:image missing-required-key}, :diversity {:image missing-required-key}, :product {:image missing-required-key}, :team {:image missing-required-key}, :values {:image missing-required-key}, :finances {:image missing-required-key}, :help {:image missing-required-key}, :growth {:image missing-required-key}, :marketing {:image missing-required-key}}
```

**Comment by martinklepsch** [comment]
huh, that's strange. I removed the `:image` key from the sections schema so I'm not sure why that shows up. You're running the referenced API branch, yes?

**Comment by bago2k4** [comment]
yup, i'm.

**Comment by bago2k4** [comment]
closing this since i merged it in my current branch: dashboard-su-manage

**Pull request closed without merging**